### PR TITLE
TE-443 first working version of "generate" mojo

### PR DESCRIPTION
**Pull request by franzbecker**

### CODE REVIEW

File: src/main/java/org/testeditor/maven/generate/GenerateMojo.java [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+			dependencies.add(dependency("org.testeditor", "gradle-tooling-api", testEditorVersion));
+			dependencies.add(dependency("org.testeditor", "org.testeditor.dsl.common.model", testEditorVersion));
+		}
+		// required for all versions
+		dependencies.add(dependency("org.testeditor", "org.testeditor.dsl.common", testEditorVersion));
+		dependencies.add(dependency("org.testeditor", "org.testeditor.aml.model", testEditorVersion));
+		dependencies.add(dependency("org.testeditor", "org.testeditor.aml.dsl", testEditorVersion));
+		dependencies.add(dependency("org.testeditor", "org.testeditor.tsl.model", testEditorVersion));
+		dependencies.add(dependency("org.testeditor", "org.testeditor.tsl.dsl", testEditorVersion));
+		dependencies.add(dependency("org.testeditor", "org.testeditor.tml.model", testEditorVersion));
```
Comment (gunther-bachmann) [inline src/main/java/org/testeditor/maven/generate/GenerateMojo.java]: tml not existent for minor >= 2

File: src/main/java/org/testeditor/maven/generate/GenerateMojo.java [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+		}
+		return element("sourceRoots", sourceRoots.toArray(new Element[0]));
+	}
+
+	// @formatter:off
+	private Element getLanguages() {
+		return element("languages",
+			element("language", element("setup", "org.testeditor.aml.dsl.AmlStandaloneSetup")),
+			element("language", element("setup", "org.testeditor.tsl.dsl.TslStandaloneSetup")),
+			element("language", element("setup", "org.testeditor.tml.dsl.TmlStandaloneSetup")),
```
Comment (gunther-bachmann) [inline src/main/java/org/testeditor/maven/generate/GenerateMojo.java]: tml not existent in minor >= 2

File: src/test/java/org/testeditor/maven/generate/integration/GenerateMojoIntegrationTest.xtend [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+					<url>http://dl.bintray.com/test-editor/maven</url>
+				</pluginRepository>
+			</pluginRepositories>
+		
+			<build>
+				<plugins>
+					<plugin>
+						<groupId>org.testeditor</groupId>
+						<artifactId>testeditor-maven-plugin</artifactId>
+						<version>1.0-SNAPSHOT</version>
```
Comment (gunther-bachmann) [inline src/test/java/org/testeditor/maven/generate/integration/GenerateMojoIntegrationTest.xtend]: shouldn't this version correspond to our test editor version?

Comment (franzbecker) [inline src/test/java/org/testeditor/maven/generate/integration/GenerateMojoIntegrationTest.xtend]: No as this plugin should be independent and (hopefully) stable across test editor versions.

**Pull request merged**